test/e2e/fetch-changed-files.js

```diff
@@ -0,0 +1,33 @@
+const axios = require('axios');
```

Comment:
I'm not sure, but we might be able to avoid a fetch to github here, and instead just use git. Not sure if we have the repo and git history in the right time and place on CI to do it, but it would be good if we could: more efficient than a network request, and not dependent on network conditions or the github api possibly being down.

Reply:
thank you Dan! That is a good point. I have an alternative branch where I explored a bit the git ci option, however it was not straight-forward and I believe we might need to tweak some things on the config.yml file in order to accomplish it. I decided to go for this option to not over-engineer it, but for the reason you mention it might be worth to try the ci option a bit further.



https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/80853/workflows/9ac00125-66fc-400d-82ac-3f22b4c928c1/jobs/2859112

Reply:
I didn't try but something like this might work?

``
gitDiff = await git.diffSummary(['--name-only',  `origin/develop...${process.env.CIRCLE_SHA1}`, 'test/**/*.spec.js']);
```

https://github.com/MetaMask/metamask-extension/blob/develop/development/generate-rc-commits.js#L2

https://github.com/steveukx/git-js?tab=readme-ov-file#git-diff

Reply:
Does the local git command like that work if you're doing a shallow checkout?  Also, the way this is written, it will be hitting the GitHub API hundreds of times per workflow (because there are hundreds of parallel machines running the workflow).  Perhaps do it in `prep-deps`, and then `persist_to_workspace`?

Reply:
Good point on the shallow checkout. Could be addressed by a `git fetch --negotiation-tip=develop origin e23f55d80e2c97d56f4fd929873b87e7f465b351` (possibly in `prep-deps`)?

Reply:
thank you for the suggestions :pray: :heart:  In the shallow clone we do this:

```
 git clone --depth 1 --no-checkout "$CIRCLE_REPOSITORY_URL" .
 git fetch --depth 1 origin "$CIRCLE_SHA1"  
```
I was thinking we would need to do git fetch develop explicitly to get available the develop branch commits in the ci environment? :thinking:  I'm also not sure if we would need to increase the depth too here :thinking: I can investigate this further

Reply:
logic for git fetch with depth and git diff has been added

Reply:
closing as this is now resolved :+1: